### PR TITLE
Added path limiting to client module loading.

### PR DESCRIPTION
**Pull request by NouberNou**
Client plugins are now limited to being loaded from specific mod folders, in which they will have a folder called intercept. That folder will contain plugin dlls. Plugins are referred to by their dll name, minus the file extension.

For example:

`"intercept" callExtension "load_extension:example_dll";`

### CODE REVIEW

**Comment by NouberNou** [comment]
My only concern with this is that I ran into a very weird race condition with easylogging++ after using the PBO search tool. I've never seen this in other projects that use this code (ACE VD and ACRE, which has been using it in lots of test deployments for months now). I fixed it by removing a logging call from the constructor of the extensions module, but I am not sure if that actually means anything (it is also where the search happens, but it just seemed odd).

The search code is using a really hacked out NtDll loaded functions search. @jaynus wrote it for ACE VD, and it is mostly black magic Windows API shit to me.

File: rv/addons/core/boot.sqf [hunk 1]
```diff
@@ -48,5 +48,24 @@ _res = "intercept" callExtension format["init_patch:%1", (productVersion select
     str parsingNamespace;
     "intercept" callExtension "invoker_end_register:";
 // };
-//"intercept" callExtension "load_extension:z\intercept\build\win32\example_frag\RelWithDebInfo\example_frag.dll";
+//"intercept" callExtension "load_extension:example_dll";
+
+_intercept_projects = configFile >> "Intercept";
+for "_i" from 0 to (count _intercept_projects)-1 do {
```
Comment (jokoho48) [inline rv/addons/core/boot.sqf]: why you using here not count/forEach and configClasses? that is faster than for

Comment (NouberNou) [inline rv/addons/core/boot.sqf]: No real reason, this runs at boot once, so it being super performant is not really that critical.

**Comment by jaynus** [comment]
+1 for win32 internals voodoo

**Comment by NouberNou** [comment]
Yea but you were right, it beats trying to parse every manner of fucking mod path loading.

Though someone will have to write a linux version of this at some point.

**Comment by jaynus** [comment]
That's WAY easier - you just read /proc/<pid>/fd

Yea but you were right, it beats trying to parse every manner of fucking
mod path loading.

Though someone will have to write a linux version of this at some point.

—
Reply to this email directly or view it on GitHub
<https://github.com/intercept/intercept/pull/95#issuecomment-186072367>.

**Pull request merged**